apps/ewallet_db/lib/ewallet_db/user_backup_code.ex

```diff
@@ -0,0 +1,139 @@
+# Copyright 2018-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EWalletDB.UserBackupCode do
+  @moduledoc """
+  Ecto Schema representing the backup codes of the user.
+  """
+  use Ecto.Schema
+  use Utils.Types.ExternalID
+  import Ecto.Changeset
+  import Ecto.Query, only: [from: 2]
+  alias Ecto.UUID
+  alias Ecto.Multi
+  alias EWalletDB.{User, UserBackupCode, Repo}
+
+  @primary_key {:uuid, UUID, autogenerate: true}
+
+  schema "user_backup_code" do
+    field(:hashed_backup_code, :string)
+    field(:used_at, :naive_datetime_usec)
+
+    belongs_to(
+      :user,
+      User,
+      foreign_key: :user_uuid,
+      references: :uuid,
+      type: UUID
+    )
+
+    timestamps()
+  end
+
+  defp changeset(%UserBackupCode{} = user_backup_code, attrs) do
+    user_backup_code
+    |> cast(attrs, [:hashed_backup_code, :user_uuid, :used_at])
+    |> validate_required([:hashed_backup_code, :user_uuid])
+    |> unique_constraint(:user_backup_code)
+    |> assoc_constraint(:user)
+  end
+
+  @doc """
+  Insert multiple user_backup_code within a single transaction by breaks the hashed_backup_codes into multiple record.
+  """
+  def insert_multiple(%{user_uuid: nil}), do: {:error, :invalid_parameter}
+
+  def insert_multiple(%{hashed_backup_codes: hashed_backup_codes} = attrs)
+      when is_list(hashed_backup_codes) do
+    attrs
+    |> build_multiple_attrs()
+    |> map_changeset()
+    |> do_insert()
+  end
+
+  defp do_insert(changesets) when is_list(changesets) do
+    changesets
+    |> Enum.with_index()
+    |> Enum.reduce(Multi.new(), fn {changeset, index}, acc ->
+      Multi.insert(acc, :"ubc_#{index}", changeset)
```

Comment:
`ubc_#{index}` is quite hard to guess what it stands for. Can we use its full name of the action? e.g. `insert_user_backup_code_#{index}`?